src/utils/thread_helper.cpp

```diff
@@ -1343,25 +1377,86 @@ otError ThreadHelper::RetrieveTelemetryData(Mdns::Publisher *aPublisher, threadn
+                           sizeof(mNat64PdCommonSalt));
 
                     sha256.Start();
                     sha256.Update(ipAddrShaInput, sizeof(ipAddrShaInput));
                     sha256.Finish(hash);
 
                     nat64Mapping->mutable_hashed_ipv6_address()->append(reinterpret_cast<const char *>(hash.GetBytes()),
-                                                                        sizeof(hash.GetBytes()));
+                                                                        Sha256::Hash::kSize);
                     // Remaining time is not included in the telemetry
                 }
             }
         }
         // End of Nat64Mapping section.
 #endif // OTBR_ENABLE_NAT64
+#if OTBR_ENABLE_DHCP6_PD
+        // Start of Dhcp6PdState section.
+        {
+            wpanBorderRouter->set_dhcp6_pd_state(
+                Dhcp6PdStateFromOtDhcp6PdState(otBorderRoutingDhcp6PdGetState(mInstance)));
+        }
+        // End of Dhcp6PdState section.
 
-        // End of WpanBorderRouter section.
+        // Start of Hashed PD prefix
+        {
+            otBorderRoutingPrefixTableEntry aPrefixInfo;
+            const uint8_t                  *prefixAddr          = nullptr;
+            const uint8_t                  *truncatedHash       = nullptr;
+            constexpr size_t                kHashPrefixLength   = 6;
+            constexpr size_t                kHashedPrefixLength = 2;
+            std::vector<uint8_t>            hashedPpdHeader     = {0x20, 0x01, 0x0d, 0xb8};
+            std::vector<uint8_t>            hashedPdTailer      = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+            std::vector<uint8_t>            hashedPdPrefix;
+            hashedPdPrefix.reserve(16);
+            Sha256       sha256;
+            Sha256::Hash hash;
+
+            otBorderRoutingGetPdOmrPrefix(mInstance, &aPrefixInfo);
+            prefixAddr = aPrefixInfo.mPrefix.mPrefix.mFields.m8;
+
+            sha256.Start();
+            sha256.Update(prefixAddr, kHashPrefixLength);
+            sha256.Update(mNat64PdCommonSalt, kNat64PdCommonHashSaltLength);
+            sha256.Finish(hash);
+
+            // Append hashedPpdHeader
+            hashedPdPrefix.insert(hashedPdPrefix.end(), hashedPpdHeader.begin(), hashedPpdHeader.end());
+
+            // Append the first 2 bytes of the hashed prefix
+            truncatedHash = hash.GetBytes();
+            hashedPdPrefix.insert(hashedPdPrefix.end(), truncatedHash, truncatedHash + kHashedPrefixLength);
+
+            // Append ip[6] and ip[7]
+            hashedPdPrefix.push_back(prefixAddr[6]);
+            hashedPdPrefix.push_back(prefixAddr[7]);
+
+            // Append hashedPdTailer
+            hashedPdPrefix.insert(hashedPdPrefix.end(), hashedPdTailer.begin(), hashedPdTailer.end());
+
+            // Convert the vector to a string
+            std::string result(hashedPdPrefix.begin(), hashedPdPrefix.end());
```

Comment:
Variable declaration should be grouped together at the beginning of the code block.

BTW, this variable is not so necessary?  Probably you can do this:
```
wpanBorderRouter->set_hashed_pd_prefix(std::string(hashedPdPrefix.begin(), hashedPdPrefix.end()));
```

Reply:
Thanks!